tests/ut_mdesktopentry.cpp

```diff
@@ -284,9 +290,13 @@ void UtMDesktopEntry::values()
     } else if (key == "Hidden") {
         QCOMPARE(entry.hidden() ? QString("true") : QString("false"), expected);
     } else if (key == "OnlyShowIn") {
+#if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
```

Comment:
Why not just migrate the below QRegExp to QRegularExpression on this and following cases? There's couple cases needing version specific lines for SkipEmptyParts but that's less variation and would be nice to have qt6 also executing the tests.

Reply:
This does seem to work on my end with Qt6 at least on top of this `qt6` branch rebased on `master`:
```diff
--- a/tests/ut_mdesktopentry.cpp
+++ b/tests/ut_mdesktopentry.cpp
@@ -290,12 +290,18 @@ void UtMDesktopEntry::values()
     } else if (key == "Hidden") {
         QCOMPARE(entry.hidden() ? QString("true") : QString("false"), expected);
     } else if (key == "OnlyShowIn") {
+        QCOMPARE(entry.onlyShowIn(), expected.split(QRegularExpression("\\s*;\\s*"),
 #if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
-        QCOMPARE(entry.onlyShowIn(), expected.split(QRegExp("\\s*;\\s*"), QString::SkipEmptyParts));
+                                                    QString::SkipEmptyParts));
+#else
+                                                    Qt::SkipEmptyParts));
 #endif
     } else if (key == "NotShowIn") {
+        QCOMPARE(entry.notShowIn(), expected.split(QRegularExpression("\\s*;\\s*"),
 #if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
-        QCOMPARE(entry.notShowIn(), expected.split(QRegExp("\\s*;\\s*"), QString::SkipEmptyParts));
+                                                    QString::SkipEmptyParts));
+#else
+                                                    Qt::SkipEmptyParts));
 #endif
     } else if (key == "TryExec") {
         QCOMPARE(entry.tryExec(), expected);
@@ -306,12 +312,18 @@ void UtMDesktopEntry::values()
     } else if (key == "Terminal") {
         QCOMPARE(entry.terminal() ? QString("true") : QString("false"), expected);
     } else if (key == "MimeType") {
+        QCOMPARE(entry.mimeType(), expected.split(QRegularExpression("\\s*;\\s*"),
 #if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
-        QCOMPARE(entry.mimeType(), expected.split(QRegExp("\\s*;\\s*"), QString::SkipEmptyParts));
+                                                    QString::SkipEmptyParts));
+#else
+                                                    Qt::SkipEmptyParts));
 #endif
     } else if (key == "Categories") {
+        QCOMPARE(entry.categories(), expected.split(QRegularExpression("\\s*;\\s*"),
 #if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
-        QCOMPARE(entry.categories(), expected.split(QRegExp("\\s*;\\s*"), QString::SkipEmptyParts));
+                                                    QString::SkipEmptyParts));
+#else
+                                                    Qt::SkipEmptyParts));
 #endif
     } else if (key == "StartupNotify") {
         QCOMPARE(entry.startupNotify() ? QString("true") : QString("false"), expected);
```

Reply:
Makes sense. Just could do the SkipEmptyParts check with < 5.14 as it was introduced already there.

This PR has been pending a year waiting for updates. Would you feel like doing a continuation PR with those? I added a couple small remarks more here, but all remaining work should be quite mechanical.